test/ModuleInterface/existential-any.swift

```diff
@@ -38,3 +38,11 @@ public struct S {
   // CHECK: public var q: any main.Q
   public var q: any Q
 }
+
+
+public protocol ProtocolTypealias {
+  typealias A = P
+}
+
+// CHECK: public func dependentExistential<T>(value: (T) -> any main.P) where T : main.ProtocolTypealias
+public func dependentExistential<T: ProtocolTypealias>(value: (T) -> T.A) {}
```

Comment:
My only hesitation here is that we might want to think about whether this interferes with the ability of `-preserve-types-as-written` to help library owners work around naming ambiguities. There are already places where library owners can't control the type names that are written into the interface, though, so maybe it's not much of a regression to lack control for this kind of signature.

Reply:
In any case, this isn't a long term fix; either we need to figure out how to make `any` with protocol typealiases work (seems unlikely), or eventually ban them in source.